src/mongocxx/test_util/client_helpers.cpp

```diff
@@ -389,42 +389,24 @@ std::string tolowercase(stdx::string_view view) {
 }
 
 void check_outcome_collection(mongocxx::collection* coll, bsoncxx::document::view expected) {
-    std::vector<std::string> actual_data;
-    std::vector<std::string> expected_data;
-
     read_concern rc;
     rc.acknowledge_level(read_concern::level::k_local);
     auto old_rc = coll->read_concern();
     coll->read_concern(rc);
 
     options::find options{};
-    builder::basic::document sort{};
-    sort.append(builder::basic::kvp("_id", 1));
-    options.sort(sort.extract());
-
-    std::vector<document::view> expected_docs{};
-    for (auto&& ele : expected["data"].get_array().value) {
-        expected_docs.push_back(ele.get_document().value);
-    }
-
-    auto it = expected_docs.begin();
-
-    for (auto&& doc : coll->find({}, options)) {
-        if (it == expected_docs.end()) {
-            FAIL("found more documents than expected.");
-        }
-
-        auto next_expected = *it;
-        it++;
+    options.sort(make_document(kvp("_id", 1)));
 
-        REQUIRE_BSON_MATCHES(doc, next_expected);
-    }
+    auto actual = coll->find({}, options);
+    std::equal(std::begin(actual),
+               std::end(actual),
+               std::begin(expected["data"].get_array().value),
+               [](document::view doc, bsoncxx::array::element ele) {
+                   REQUIRE_BSON_MATCHES(doc, ele.get_document().value);
+                   return true;
+               });
```

Comment:
During my debugging efforts, I had a difficult time trying to parse what was happening here. I realized all this did was assert that two lists contained the same elements and had the same length. `REQUIRE_BSON_MATCHES` will show mismatches in the debugging output.

Reply:
Updated this in the last commit. Since this checks from `begin(range1)` to `end(range1)` and iterates `range2` this would've returned `true` if `range1` was empty. I swapped the ranges so we check at least the length of `expected_data`. Afterward actual should be empty since it's a cursor if they're the same length.

Reply:
Also, I removed the `write_concern` swap logic and ran it evergreen ([here](https://evergreen.mongodb.com/version/5f11c0d1850e61351dc5975f)). All of the tests pass so I'm not sure if we still need it.

Reply:
Ah, nevermind. After investigating CDRIVER-2901 it seems better to leave it in.

Reply:
This looks a lot cleaner. And I agree, using local read concern seems necessary.